src/topology.jl

```diff
@@ -52,34 +95,71 @@ function Cart_get(comm::Comm, maxdims::Integer)
     return dims, periods, coords
 end
 
+@deprecate Cart_get(comm, maxdims) Cart_get(comm)
+
 """
-    Cartdim_get(comm::Comm)
+    ndims = Cartdim_get(comm::Comm)
 
 Return number of dimensions of the Cartesian topology associated with the communicator `comm`.
+
+# External links
+$(_doc_external("MPI_Cartdim_get"))
 """
 function Cartdim_get(comm::Comm)
-    dims    = Cint[0]
+    dims = Ref{Cint}()
     # int MPI_Cartdim_get(MPI_Comm comm, int *ndims)
     @mpichk ccall((:MPI_Cartdim_get, libmpi), Cint,
                   (MPI_Comm, Ptr{Cint}),
                   comm, dims)
-    return Int(dims[1])
+    return Int(dims[])
 end
 
-function Cart_coords!(comm::Comm, rank::Integer, maxdims::Integer, coords::MPIBuffertype{Cint})
+"""
+    Cart_coords!(comm::Comm, rank::Integer, coords)
+
+Determine coordinates of a given process in Cartesian communicator.
+
+# External links
+$(_doc_external("MPI_Cart_coords"))
+"""
+function Cart_coords!(comm::Comm, rank::Integer, coords::MPIBuffertype{Cint})
+    maxdims = Cartdim_get(comm)
     # int MPI_Cart_coords(MPI_Comm comm, int rank, int maxdims, int coords[])
     @mpichk ccall((:MPI_Cart_coords, libmpi), Cint,
                   (MPI_Comm, Cint, Cint, Ptr{Cint}),
                   comm, rank, maxdims, coords)
 end
 
-function Cart_coords(comm::Comm, maxdims::Integer)
+"""
+    Cart_coords(comm::Comm)
+
+Determine coordinates of local process in Cartesian communicator.
+
+See also [`Cart_coords!`](@ref).
+"""
+function Cart_coords(comm::Comm)
+    maxdims = Cartdim_get(comm)
     ccoords = Vector{Cint}(undef, maxdims)
     rank    = Comm_rank(comm)
-    Cart_coords!(comm, rank, maxdims, ccoords)
+    Cart_coords!(comm, rank, ccoords)
     Int.(ccoords)
 end
 
+@deprecate(Cart_coords!(comm, rank, maxdims::Integer, coords),
```

Comment:
Can you move the deprecated calls to `deprecated.jl`?

Reply:
Ohh right, I missed that somehow.

Reply:
Done in https://github.com/JuliaParallel/MPI.jl/pull/340/commits/8e8e0ed08f721b6294bdd88ee6ccea07dbeac1ab

Reply:
Oops, sorry, I accidentally started a couple of reviews by pasting the commit url...